Move Proof Mode and SSR chapter one level up. (#20990)
rocq-prover/rocq: doc/sphinx/proof-engine/ssreflect-proof-language.rst

of general interest and is available in Rocq starting from version 8.7.

|SSR| was developed independently of the tactics described in
:ref:`tactics`. Indeed the scope of the tactics part of |SSR| largely
overlaps with the standard set of tactics. Eventually the overlap will
be reduced in future releases of Rocq.

Proofs written in |SSR| typically look quite different from the

> ones written using only tactics as per :ref:`tactics`. We try to

Reviewer: ```suggestion
ones written using only tactics described in :ref:`tactics`. We try to
```

summarise here the most “visible” ones in order to help the reader

> already accustomed to the tactics described in :ref:`tactics` to read

Reviewer: ```suggestion
already accustomed to the tactics in :ref:`tactics` to read
```
